packages/web-api-scan-runner/src/combined-result/combined-results-blob-provider.ts

```diff
@@ -5,35 +5,31 @@ import { GuidGenerator } from 'common';
 import { inject, injectable } from 'inversify';
 import { GlobalLogger } from 'logger';
 import { CombinedScanResultsProvider, CombinedScanResultsReadResponse } from 'service-library';
-
-export interface CombinedResultsBlobInfo {
-    response: CombinedScanResultsReadResponse;
-    blobId: string;
-}
+import { CombinedResultsBlob } from '../types/combined-results-blob';
 
 @injectable()
-export class CombinedResultsBlobGetter {
+export class CombinedResultsBlobProvider {
     public constructor(
-        @inject(GlobalLogger) private readonly logger: GlobalLogger,
-        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
         @inject(CombinedScanResultsProvider) protected readonly combinedScanResultsProvider: CombinedScanResultsProvider,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(GlobalLogger) private readonly logger: GlobalLogger,
     ) {}
 
-    public async getBlobInfo(givenResultsBlobId: string): Promise<CombinedResultsBlobInfo> {
-        let returnedResultsBlobId = givenResultsBlobId;
+    public async getBlob(existingCombinedResultsBlobId: string): Promise<CombinedResultsBlob> {
```

Comment:
Nit: since we expect `existingCombinedResultsBlobId` to sometimes be undefined, could it either be nullable or have type string | undefined?

Reply:
updated